### ISSUE TITLE
Different help messages for same problem

### ISSUE THREAD

**Issue by marimeireles** [mamba-org/mamba#456]
So, if you don't have a `target_prefix` and you need to activate one you might get one of the two messages depending on what you're doing:

https://github.com/TheSnakePit/mamba/blob/54413e75986e16ea187789d9ab1ad47c28f3da6c/src/main.cpp#L252-L253

https://github.com/TheSnakePit/mamba/blob/54413e75986e16ea187789d9ab1ad47c28f3da6c/src/main.cpp#L413

One says it's necessary to add a `-p` and the other doesn't. Which is right?

**Comment by marimeireles** [comment]
I think without the `-p` but it might depend on the context?

**Comment by hugobuddel** [comment]
I'm very confused:
```
~$ ./micromamba --version
0.4.2

~$ ./micromamba create -p testenv5
[...]
Transaction

  Prefix: testenv5

  Nothing to remove

Transaction starting
Transaction finished

~$ ls testenv5/
conda-meta/  pkgs/

~$ ./micromamba activate -p testenv5
The following arguments were not expected: testenv5 -p activate
Run with --help for more information.

~$ ./micromamba activate testenv5
The following arguments were not expected: testenv5 activate
Run with --help for more information.
```
How to activate an environment?

**Comment by marimeireles** [comment]
Hey @hugobuddel, maybe this other issue helps: https://github.com/TheSnakePit/mamba/issues/457
Or even the configurations for the CI to run: 
https://github.com/TheSnakePit/mamba/blob/54413e75986e16ea187789d9ab1ad47c28f3da6c/.github/workflows/main.yml#L36-L43
That was actually what got me rolling.

**Comment by hugobuddel** [comment]
Oh that is smart @marimeireles, looking at the CI scripts. I was already wondering where you got those commands from in #457. Ah, I think I get it now, `mamba.sh` creates a bash function called `micromamba`. So you are not supposed to run the `micromamba` executable directly, but only through this function. This is great. Sorry for polluting this issue.

**Comment by marimeireles** [comment]
Oh! No problem, glad I could help :)

**Comment by adriendelsalle** [comment]
Hi @hugobuddel !

A lot has been fixed, improved and released since (we are at 0.13.1). Hopefully this case is fixed now.
Let me try to advertise a bit the behavior, to serve community similar issues:
- `./micromamba` is calling directly the executable
  - after running `./micromamba shell init -s bash -p ~/micromamba`, calling `micromamba` is in fact a shell function allowing extra operations that calling an executable can't do (such as updating your shell `$PATH`)
  - while `micromamba` shell function has a `activate` sub-command, the executable has not: it relies on `./micromamba shell activate -p <prefix>`
    - if you use it directly, you will probably notice in some cases that the environment/prefix is not properly activated then
    - you should rely on the shell function to do activation, it's the behavior you can see in the link kindly posted by @marimeireles 
- the output `The following arguments were not expected: testenv5 -p activate` was misleading, because the command arguments are reverted during parsing
  - the CLI would print you `The following argument was not expected: activate` now
- the [2nd link](https://github.com/TheSnakePit/mamba/blob/54413e75986e16ea187789d9ab1ad47c28f3da6c/src/main.cpp#L413) you posted was either wrong, either working at that time but not working anymore. The help footer has been updated accordingly!

I'm closing it but feel free to re-open or open a new one if needed!

**Comment by hugobuddel** [comment]
Thanks for updating the issue. I'll give micromamba another try soon.
